vorlesung14/zusammenfassung14.md

```diff
@@ -0,0 +1,408 @@
+# Kapitel 14 Implementierung und Dokumentation 
+
+**Autor: **  
+André Matutat
+
+
+
+## Lernziele
+
+- Wissen, was die Ziele der Implementierung sind
+- Verstehen, wie UML-Diagramme in Software umgesetzt werden
+- Code Smells und schlechtes Exception Handling identifizieren und beheben können
+
+## Ziel der Implementierung 
+
+- Ausgangspunkt: Entwurfsspezifikation mit System- und Softwarearchitektur
+- Transformation der Entwurfsergebnisse in Software, die auf einem bestimmten Zielrechner ausführbar ist
+- Konsistenz mit dem Entwurf: Strukturen des Entwurfs lassen sich im Source Code leicht wieder finden
+- Schnittstellen des Programmsystems sind explizit beschrieben
+- Ergebnis: Software, Dokumentation, Entwicklertests
+
+**Die Erfüllung der Eigenschaften ist abhängig von**
+- der Art der Programmiersprache und Programmierkonzepte (z.B. objektorientiert)
+- Der kompetentz der Beteiligten (z.B. des Software-Entwicklers)
+
+## Übersetzen von UML Klassendiagramme in objektorientierte Programmiersprachen 
+
+![](vorlesung14/bilder/bild1.png)
+- Bei Übersetzung in Source-Code auf Vererbung und Sichtbarkeit achten
+- Es kann vereinbart werden, dass get- und set-Methoden in Klassendiagrammen weggelassen werden
```

Comment:
Bezeichnet man die nicht eher als getter und setter Methoden?

Reply:
Ich glaube "getter" und "setter" sind ehr Slang Begriffe. 
Für die tighten Coder Gangs die sich in der Hood  noch respect GETTEN müssen.